python/kvikio/kvikio/_lib/arr.pyx

```diff
@@ -130,36 +130,37 @@ cdef class Array:
                 self.strides_mv = None
         else:
             PyObject_GetBuffer(obj, &pybuf, PyBUF_FULL_RO)
-
-            if pybuf.suboffsets != NULL:
-                raise NotImplementedError("Suboffsets are not supported")
-
-            self.ptr = <uintptr_t>pybuf.buf
-            self.obj = pybuf.obj
-            self.readonly = <bint>pybuf.readonly
-            self.ndim = <Py_ssize_t>pybuf.ndim
-            self.itemsize = <Py_ssize_t>pybuf.itemsize
-
-            if self.ndim > 0:
-                self.shape_mv = new_Py_ssize_t_array(self.ndim)
-                memcpy(
-                    &self.shape_mv[0],
-                    pybuf.shape,
-                    self.ndim * sizeof(Py_ssize_t)
-                )
-                if not PyBuffer_IsContiguous(&pybuf, b"C"):
-                    self.strides_mv = new_Py_ssize_t_array(self.ndim)
+            try:
+                if pybuf.suboffsets != NULL:
+                    raise NotImplementedError("Suboffsets are not supported")
```

Comment:
For example here is one exception we raise

Though it is also possible to wind up with an exception due to some bug in the code below

In any event we want to make sure normal resource cleanup can proceed as usual (without potentially causing segfaults)